src/middlewares/controllerHandler.js

```
@@ -0,0 +1,63 @@
+      const error = new Error('Authentication required');
+      error.statusCode = 401;
+      error.status = 'fail';
+      error.isOperational = true;
+      error.errorCode = 'UNAUTHORIZED';
+      throw error;
+    }
+
+    // Set auth in context for the controller to use
+    c.set('userId', auth.userId);
+
+    return await controllerFn(c);
```

ESLint

[error] Redundant use of `await` on a return value.

(no-return-await)